tools/taskcluster/execute-firebase-test.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This script installs google cloud sdk, logs into google firebase, executes tests,
+# and collects test artifacts into the test_artifacts folder
+
+# If a command fails then do not proceed and fail this script too.
+set -ex
+#########################
+# The command line help #
+#########################
+display_help() {
+    echo "Usage: $0 Build_Variant Apk_Name Device_Config [Device_Config...]"
+    echo
+    echo "Do not put .apk prefix in Apk_Name"
+    echo "Device_Config should be in landscape mode (e.g. model=taimen,version=26,orientation=landscape)"
+    echo
+}
+
+# Basic parameter check
+if [[ $# -lt 3 ]]; then
+    echo "Your command line contains $# arguments"
+    display_help
+    exit 1
+fi
+
+# Get test configuration
+config_count=`expr $# - 2`
+config_param="--device $3"
+for ((i=4; i <= $#; i++))
+{
+    config_param+=" --device ${!i}"
+}
+
+# From now on disable exiting on error. If the tests fail we want to continue
+# and try to download the artifacts. We will exit with the actual error code later.
+set +e
+
+# Execute test set
+./google-cloud-sdk/bin/gcloud --format="json" firebase test android run \
+--type instrumentation \
+--app ./app/build/outputs/apk/$1/$2.apk \
+--test ./app/build/outputs/apk/androidTest/$1/$2-androidTest.apk \
+--use-orchestrator \
+--environment-variables clearPackageData=true \
+--results-bucket firefox-tv_test_artifacts \
+--timeout 30m \
+--no-auto-google-login $config_param
```

Comment:
This block seems like something we will probably want to share in both Firebase and other tests.  I'm not terribly familiar with Shell scripting, is there any way to break this out and reuse it?

Reply:
This is google gcloud sdk cli command, and I have thought about creating a wrapper method for it, but for each project, most parameters would be slightly different - so it would be more clear to the user if the actual gcloud test request is readily visible.